app/features/onboarding/onboarding-actions.server.ts

```diff
@@ -17,35 +15,33 @@ import {
   onboardingOrganizationSchema,
   onboardingUserProfileSchema,
 } from './onboarding-client-schemas';
-import { withUserRequiringOnboarding } from './onboarding-middleware.server';
+import { requireUserNeedsOnboarding } from './onboarding-helpers.server';
+
+export const onboardingUserProfileAction = async ({
+  request,
+}: Pick<ActionFunctionArgs, 'request'>) => {
+  const user = await requireUserNeedsOnboarding(request);
```

Comment:
Also, is `requireUserNeedsOnboarding` efficient aka. parallel? We probably want to make sure it is and if NOT refactor it to be that in a different PR.